DragAndDrop/src/main/java/com/microsoft/device/display/samples/draganddrop/fragment/DropTargetFragment.kt

```diff
@@ -6,206 +6,210 @@
-                binding.dropImageHint.updateColorAndElevation(null, DEFAULT_ELEVATION)
-                setElevation(
-                    DEFAULT_ELEVATION,
-                    binding.imageHintImageView,
-                    binding.imageHintTextView
-                )
-            }
-
-            MimeType.TEXT_PLAIN -> {
-                binding.textHintImageView
-                binding.dropTextHint.updateColorAndElevation(null, DEFAULT_ELEVATION)
-                setElevation(DEFAULT_ELEVATION, binding.textHintImageView, binding.textHintTextView)
-            }
+    private fun handleImageDrop(item: ClipData.Item) {
+        item.uri?.let { uri ->
+            val bitmap =
+                BitmapFactory.decodeStream(requireContext().contentResolver.openInputStream(uri))
+            setImageTarget(bitmap)
+        } ?: run {
+            Log.e(TAG, "Clip data is missing it's URI")
         }
     }
 
-    /**
-     * Adds the text from the [DragEvent.getClipData] to the drop zone
-     *
-     * @param event The [DragEvent] containing the text that will be added to the drop zone
-     */
-    private fun handleTextDrop(event: DragEvent) {
-        val item = event.clipData.getItemAt(0)
-        val dragData = item.text.toString()
-        var sourceView = event.localState as? View
-        sourceView?.let {
-            it.remove()
-        } ?: run {
-            sourceView = TextView(requireContext()).apply {
-                text = dragData
-            }
+    private fun getTextFromClipData(item: ClipData.Item): String {
+        val parcelFileDescriptor: ParcelFileDescriptor? = try {
+            requireActivity().contentResolver.openFileDescriptor(item.uri, "r")
+        } catch (e: FileNotFoundException) {
+            e.printStackTrace()
+            null
         }
 
-        removeViews(binding.textHintImageView, binding.textHintTextView)
-        binding.emptyText.replaceWith(sourceView)
-        sourceView?.setOnLongClickListener(null)
-        binding.dropTextHint.setOnDragListener(null)
-    }
+        if (parcelFileDescriptor == null) {
+            Log.e(TAG, "Invalid File Descriptor")
+            return ""
+        }
 
-    /**
-     * Adds the image from the [DragEvent.getClipData] to the drop zone
-     *
-     * @param event The [DragEvent] containing the image that will be added to the drop zone
-     */
-    private fun handleImageDrop(event: DragEvent) {
-        val item = event.clipData.getItemAt(0)
-        var sourceView = event.localState as? View
-        sourceView?.let {
-            // Remove the local image view, source view is null if drop from another app
-            it.remove()
-        } ?: run {
-            sourceView = ImageView(requireContext()).apply {
-                setDragAndDropImageURI(requireActivity(), event, item)
+        val fileDescriptor = parcelFileDescriptor.fileDescriptor
+        val reader = BufferedReader(FileInputStream(fileDescriptor).reader())
```

Comment:
Instead of this code, you can use directly File(fileName).readText(Charsets.UTF_8) or File(fileName).inputStream().readBytes().toString(Charsets.UTF_8)